### ISSUE TITLE
Training yolov5 model appears nan

### ISSUE THREAD

**Issue by xialuxi** [sail-sg/Adan#20]

**Comment by xialuxi** [comment]
Epoch   gpu_mem       box       obj       cls       kps    labels  img_size
     4/399     3.19G   0.05632    0.0556   0.01426    0.1152       129       416: 100%|███████████████████████████████████| 367/367 [06:20<00:00,  1.04s/it]
               Class     Images     Labels          P          R     mAP@.5 mAP@.5:.95: 100%|███████████████████████████████| 13/13 [00:02<00:00,  6.47it/s]
nt:  [ 390 1761 1229]
                 all        405       3380      0.596      0.517      0.565      0.229
                 hoi        405        390       0.63      0.232      0.394      0.117
              person        405       1761       0.56       0.67      0.653      0.255
                 box        405       1229      0.597      0.649      0.647      0.316

     Epoch   gpu_mem       box       obj       cls       kps    labels  img_size
     5/399     3.19G       nan       nan       nan       nan       107       416: 100%|███████████████████████████████████| 367/367 [06:16<00:00,  1.03s/it]
               Class     Images     Labels          P          R     mAP@.5 mAP@.5:.95: 100%|███████████████████████████████| 13/13 [00:00<00:00, 17.18it/s]
                 all        405          0          0          0          0          0

**Comment by XingyuXie** [comment]
Hi, @xialuxi,
I suggest using a smaller LR of Adan to train. 
BZW, it seems that most yoloVx models are trained by SGD. 
The previous research may encounter some problems in using adaptive optimizers such as Adam to train yolo models. Many issues in the repo of yoloV7 mention that Adam may not provide good results, see https://github.com/WongKinYiu/yolov7/issues/702#issue-1360788884 and https://github.com/WongKinYiu/yolov7/issues/730#issue-1364051445.

So I suggest tuning the `LR` and `weight-decay` based on yoloVx with the official Adam's setting. 
If the final result is unsatisfactory, it may cause by Adan or Adam, but because of the difference between the adaptive type optimizer and the SGD type optimizer.

If you still need further help with using Adan and parameter tuning, please don't hesitate to leave a message here. Or add my WeChat: xyxie_joy.

**Comment by xialuxi** [comment]
Adam:
Epoch   gpu_mem       box       obj       cls       kps    labels  img_size
3/399     3.22G   0.06394   0.05859   0.01641    0.1517       173       416

Epoch   gpu_mem       box       obj       cls       kps    labels  img_size
4/399     3.22G   0.06176   0.05778    0.0158    0.1367       129       416

Epoch   gpu_mem       box       obj       cls       kps    labels  img_size
5/399     3.22G   0.05997   0.05672   0.01512    0.1282       107       416

**Comment by XingyuXie** [comment]
Dear @xialuxi,
What is your hyper-parameter of Adan and Adam? Could you please paste them here? 
So I may provide a more reasonable LR, Wd, or clip for Adan.

Best

**Comment by xialuxi** [comment]
**SGD:**
lr = 0.01, momentum=0.937, weight_decay=0.0005, nesterov=True
**Adam:**
lr = 0.01, betas=(0.937, 0.999), weight_decay=0.0005
**Adan:**
lr = 0.01, betas=(0.98, 0.92, 0.99), eps=1.0e-08, weight_decay=0.0005

**Comment by XingyuXie** [comment]
@xialuxi you may try `lr = 1e-3, betas=(0.98, 0.92, 0.99), eps=1.0e-08, weight_decay=0.02` for Adan.
As usual, the LR for the adaptive optimizer should be 10-100x times smaller than the LR used in SGD. Thus we suggest `lr = 1e-3` for Adan.
The reason that we set `wd=0.02` is due to the use of decoupled weight decay.
